modules/reindex/src/main/java/org/elasticsearch/index/reindex/GetReindexTaskAction.java

```diff
@@ -0,0 +1,145 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public class GetReindexTaskAction extends ActionType<GetReindexTaskAction.Response> {
+
+    public static final GetReindexTaskAction INSTANCE = new GetReindexTaskAction();
+    public static final String NAME = "indices:data/write/reindex/get";
+
+    public GetReindexTaskAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        private final String taskId;
+
+        public Request(String taskId) {
+            this.taskId = taskId;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.taskId = in.readString();
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(taskId);
+        }
+
+        public String getTaskId() {
+            return taskId;
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        public static final ConstructingObjectParser<Response, Void> PARSER =
+            new ConstructingObjectParser<>("reindex/get_task", a -> new Response((BulkByScrollResponse) a[0],
+                (ElasticsearchException) a[1], (ScrollableHitSource.Checkpoint) a[2]));
+
+        private static final String REINDEX_RESPONSE = "response";
+        private static final String REINDEX_EXCEPTION = "exception";
```

Comment:
Looking at other APIs, there are `error` (`TaskResult`) and `failure` (`bulkFailures`, `searchFailures` in `BulkByScrollResponse` and `shardFailures` in `SearchResponse`). I did not find `exception`, but it might be lurking somewhere hidden anyway... I wonder if we should stick to `failure` here too?
If you agree, we should also do a similar change on the .reindex index and `ReindexTaskStateDoc`.